lib/beaker/dsl/install_utils/windows_utils.rb

```diff
@@ -141,6 +141,16 @@ def install_msi_on(hosts, msi_path, msi_opts = {}, opts = {})
             # therefore, this command will always exit 0 if either service is installed
             on host, Command.new("sc query puppet || sc query pe-puppet", [], { :cmdexe => true })
 
+            # (PA-514) value for PUPPET_AGENT_STARTUP_MODE should be present in
+            # registry and honored after install/upgrade.
+            reg_query_command = %Q(reg query "HKLM\\SOFTWARE\\Wow6432Node\\Puppet Labs\\PuppetInstaller" /v "RememberedPuppetAgentStartupMode" | findstr #{msi_opts['PUPPET_AGENT_STARTUP_MODE']})
```

Comment:
After a bit of archaeology from @MosesMendoza, looks like our best bet is to call `is_x86_64?` on the host (https://github.com/puppetlabs/beaker/blob/master/lib/beaker/host.rb#L269-L271), to then determine which reg key to query.

We'll make sure to run the new PR against 32-bit and 64-bit Windows hosts to verify.